### PR TITLE
feat: harden path policy and diffstat checks

### LINKED ISSUE #21: [M5][P0] Improve path policy and diffstat checks
## Goal

Make policy checks strong enough for low-risk company usage.

## Acceptance criteria

- Deny paths support glob patterns reliably.
- Max changed files is enforced.
- Max changed lines is enforced.
- Policy failure prevents push/PR.
- Policy failure writes a clear task event.

## Notes

Do not rely on prompt-only safety.

### PR DESCRIPTION
**Pull request by hey-yvan**
Closes #21

## Summary
- Adds a dependency-free `internal/policy` package with a robust glob matcher (`**`, `*`, `?`) and structured `Violation` values for `deny_path`, `not_allowed`, `max_changed_files`, `max_changed_lines`.
- `workspace.Diffstat` now uses `git add --intent-to-add --all` + `git diff --numstat HEAD` so untracked files and total added+deleted lines are observable, not just modified file names.
- `workspace.EnforcePolicy` returns a typed `*PolicyError` carrying every violation. The worker writes a structured `policy_violation` task event (with JSON payload listing each violation) and returns the error before push/PR, so policy failures reliably block remote changes.
- `workflow.PolicyConfig` adds `max_changed_lines` while preserving the legacy `max_changed_loc` via `PolicyConfig.LineLimit()`. Existing WORKFLOW.md files keep working.
- Adds `queue.Store.AddEventWithPayload` so structured events can attach JSON details.

## Acceptance criteria
- [x] Deny paths support glob patterns reliably (covers `**`, `*`, `?`, `path/`, `**/file`, etc.)
- [x] Max changed files enforced (`max_changed_files`).
- [x] Max changed lines enforced (`max_changed_lines`, with `max_changed_loc` fallback).
- [x] Policy failure blocks push/PR (returns before `CommitAndPush` / `createPR`).
- [x] Policy failure writes a clear task event (`policy_violation` with JSON payload).

## Tests
- `internal/policy/policy_test.go` covers glob edge cases, deny/allow precedence, file/line limits, blank entries, and accumulated violations.
- `internal/workspace/manager_test.go` exercises `Diffstat` and `EnforcePolicy` end-to-end against a real temp git repo (skips if `git` is unavailable).
- `gofmt -l cmd internal`, `go vet ./...`, `go build ./...`, `go test ./...` all pass locally.

## Dependencies
- No new dependencies. Glob matching is implemented with stdlib only (no `doublestar` or similar).

### CODE REVIEW

File: internal/workspace/manager.go [hunk 1]
```diff
@@ -62,36 +64,78 @@ func RunVerify(ctx context.Context, workdir string, wf workflow.Config) error {
...
 		}
+		// numstat format: "<added>\t<deleted>\t<path>" (path may include rename "old => new")
+		parts := strings.SplitN(line, "\t", 3)
+		if len(parts) != 3 {
+			continue
+		}
+		added, _ := strconv.Atoi(parts[0])
+		deleted, _ := strconv.Atoi(parts[1])
+		d.Lines += added + deleted
+		d.Files = append(d.Files, parts[2])
```
[the remark this answers is not in the document]
Comment (hey-yvan) [reply internal/workspace/manager.go:139]: Fixed in 92041ba. Switched `Diffstat` to `git diff --numstat -z HEAD` and added a `parseNumstatZ` helper that reads the NUL-delimited records: rename entries arrive as `added\tdeleted\t\0old\0new\0`, so we record only the destination path that policy actually applies to. Added a table-driven parser test covering vanilla / `old => new` / brace `{a => b}/c` / mixed / binary, plus a regression test that does a real `git mv src/main.tf infra/main.tf` and confirms the `infra/**` deny rule now fires.
